generate-assets/src/lib.rs

```diff
@@ -310,19 +335,100 @@ fn get_license(cargo_manifest: &cargo_toml::Manifest) -> Option<String> {
+                bevy_dependency = search_bevy_in_dependencies(workspace_dependencies, bevy_crates);
+            }
+
+            if bevy_dependency.is_none() {
+                // If everything else fails, try to find any crate with a name starting with bevy
+                // This can happen if the asset depends only on past or future bevy sub-crates,
+                // or on third-party crates which name starts with bevy (might yield unaccurate results in that last case)
+                bevy_dependency = dependencies.find_map(|(_, d)| get_bevy_dependency_version(d));
+            }
+        }
+    }
+
+    bevy_dependency
+}
+
+/// Seach the first official bevy crate found in a collection of dependencies and return its version.
+/// If it was a bit more generic, this function could be called find_first_intersect_in_sorted_iterators.
+/// Both dependencies and bevy_crates are assumed to be sorted (by key for dependencies, they are in this context),
+/// and we find the first element that intersect both of them using that knowledge
+fn search_bevy_in_dependencies<'a>(
+    mut dependencies: std::collections::btree_map::Range<'a, String, cargo_toml::Dependency>,
+    mut bevy_crates: std::slice::Iter<&str>,
+) -> Option<String> {
+    let mut dependency = dependencies.next();
+    let mut bevy_crate = bevy_crates.next();
+
+    while !dependency.is_none() && !bevy_crate.is_none() {
+        let dependency_name = dependency.unwrap().0;
+        let bevy_crate_name = bevy_crate.unwrap();
+
+        if dependency_name.as_str() < *bevy_crate_name {
+            dependency = dependencies.next();
+        } else {
+            if dependency_name == bevy_crate_name {
+                let dependency_version = get_bevy_dependency_version(dependency.unwrap().1);
+                if dependency_version.is_some() {
+                    return dependency_version;
+                } else {
+                    // In this case we found an official bevy crate but we couldn't get a version from it
+                    dependency = dependencies.next();
+                    bevy_crate = bevy_crates.next();
+                }
+            } else {
+                bevy_crate = bevy_crates.next();
+            }
+        }
+    }
+    return None;
+}
+
+/// Gets the bevy version from the bevy dependency provided
+/// Returns the version number if available.
+/// If is is a git dependency, return either "main" or "git" for anything that isn't "main".
+fn get_bevy_dependency_version(dep: &cargo_toml::Dependency) -> Option<String> {
+    match dep {
+        cargo_toml::Dependency::Simple(version) => Some(version.to_string()),
+        cargo_toml::Dependency::Detailed(detail) => {
+            if let Some(version) = &detail.version {
+                Some(version.to_string())
+            } else if detail.git.is_some() {
+                if detail.branch == Some(String::from("main")) {
+                    Some(String::from("main"))
```

Comment:
I think I would like to remove this case. Targeting the "main" branch is not really a guarantee that it's tracking main.
A plugin targeting main but that wasn't updated in 10 months won't work again current main.

Reply:
I wanted to make another PR to fix #477 (and/or address https://github.com/bevyengine/bevy-assets/issues/297 ) so maybe it would be possible to use the last update date in this algorithm at the same time (with a threshold like "it needs to be updated at least less than one month ago to qualify to the main branch tag)?

It kinda make sense to at least display some version in the case where the asset is actually updated regularly to follow main.

I can remove it for now if you want.